src/applications/personalization/profile/hooks/useProfileRouteMetaData.js

```diff
@@ -0,0 +1,14 @@
+import { useLocation } from 'react-router-dom';
+import { PROFILE_PATHS_WITH_NAMES } from '../constants';
+import { getRouteInfoFromPath } from '../../common/helpers';
+
+export const useProfileRouteMetaData = () => {
+  const { pathname } = useLocation();
+  return (() => {
+    try {
+      return getRouteInfoFromPath(pathname, PROFILE_PATHS_WITH_NAMES);
+    } catch (e) {
+      return PROFILE_PATHS_WITH_NAMES.PROFILE_ROOT;
+    }
+  })();
```

Comment:
The immediately invoked function expression or 'IIFE' as they say in JS land is usually just an easy way to run a function without needing to declare it first.

In the case of the `useProfileRoutMetaData` hook, I'm looking to create an actual reusable function instead of a single use one, so thats why I didn't use an IIFE.

The IIFE was used much more in the early JS days when modules weren't a thing and es5 was still the standard. I don't use them a lot, but they can be useful for things like this, or wrapping up an async/await call easily.

Further reading:
https://developer.mozilla.org/en-US/docs/Glossary/IIFE